user/manager_test.go

```diff
@@ -1075,6 +1096,24 @@ func TestManager_Topic_Wildcard_With_Underscore(t *testing.T) {
 	require.Equal(t, ErrUnauthorized, a.Authorize(nil, "mytopicX", PermissionWrite))
 }
 
+func TestManager_WithProvisionedUsers(t *testing.T) {
+	f := filepath.Join(t.TempDir(), "user.db")
+	conf := &Config{
+		Filename:      f,
+		DefaultAccess: PermissionReadWrite,
+		ProvisionedUsers: []*User{
+			{Name: "phil", Hash: "$2a$10$YLiO8U21sX1uhZamTLJXHuxgVC0Z/GKISibrKCLohPgtG7yIxSk4C", Role: RoleAdmin},
```

Comment:
It doesn't matter if the bcrypt cost of a particular provisioned user's password hash (e.g., `$2a$XX$...`) is different from the server's configured `BcryptCost`, right?

I haven't looked close enough at the code or tested it myself, but it seems like it could be possible that the server is configured to use a cost of `10` and the provisioned hash used a cost of `12`. Then, when trying to login to the provisioned user's account, the server will generate `hash(<password>, cost: 10)` and compare it to the hash in the DB, but they wouldn't match

Reply:
Good catch. I'll go take a look.